Batteries/Data/List/Lemmas.lean

```diff
@@ -633,6 +633,43 @@ theorem dropInfix?_eq_some_iff [BEq α] {l i p s : List α} :
 @[simp] theorem dropInfix?_nil [BEq α] {s : List α} : dropInfix? s [] = some ([], s) := by
   simp [dropInfix?_eq_some_iff]
 
+/-! ### IsPrefixOf?, IsSuffixOf? -/
+
+@[simp] theorem isSome_isPrefixOf?_eq_isPrefixOf [BEq α] (xs ys : List α) :
+    (xs.isPrefixOf? ys).isSome = xs.isPrefixOf ys := by
+  match xs, ys with
+  | [], _ => simp [List.isPrefixOf?]
+  | _::_, [] => rfl
+  | _::_, _::_ =>
+    simp only [List.isPrefixOf?, List.isPrefixOf]
+    split <;> simp [*, isSome_isPrefixOf?_eq_isPrefixOf]
+
+theorem append_eq_of_isPrefixOf?_eq_some [BEq α] [LawfulBEq α] {xs ys zs : List α}
+    (h : xs.isPrefixOf? ys = some zs) : xs ++ zs = ys := by
+  match xs, ys with
+  | [], _ =>
+    simp [List.isPrefixOf?] at h
```

Comment:
I think this counts as a non-terminal simp, so should be squeezed. (Although personally I'd be happy to relax this rule a bit for `simp at`. Perhaps `simp_all` works here?

Reply:
Done.